fsspec/core.py

```diff
@@ -346,7 +346,7 @@ def _un_chain(path, kwargs):
         kws = kwargs.pop(protocol, {})
         if bit is bits[0]:
             kws.update(kwargs)
-        kw = dict(**extra_kwargs, **kws)
+        kw = dict(**{k: v for k, v in extra_kwargs.items() if k not in kws or v != kws[k]}, **kws)
```

Comment:
Given the `update` on the line above, this could be repeated updates, and that way we can be a little more explicit about the order of precedence. In your model, user-supplied arguments should always win, overriding inferred ones?

Reply:
Exactly, to completely avoid thinking about the priorities, here we simply deduplicate the same key-value pairs (so the priority is irrelevant). If the same key has two different values, they're passed as before, so that will raise the same error, e.g., `TypeError: dict() got multiple values for keyword argument 'account_name'`

`extra_kwargs = {'x': 5, 'y': 4}, kws = {'z': 4}` becomes `{'x': 5, 'y': 4, 'z': 4}`
`extra_kwargs = {'x': 5, 'y': 4}, kws = {'x': 5}` becomes `{'x': 5, 'y': 4}`
`extra_kwargs = {'x': 5, 'y': 4}, kws = {'x': 4}` becomes `dict(**{'x': 5, 'y': 4}, **{'x': 4})` and raises TypeError

Reply:
> so the priority is irrelevant

Checking the values isn't always straight forward, they might not be simple ints and str. We could catch it and pass a useful message to the user?

Reply:
Can we please include the two passing examples in some sort of test?

Reply:
Apologies for the delay.
Yes, I can add tests.